### PR TITLE
fix curses: getstr(), inch() and instr() return union type

### PR DESCRIPTION
**Pull request by Rohan-Salwan**
- fixed: #6402

### CODE REVIEW

File: stdlib/_curses.pyi [hunk 1]
```diff
@@ -444,13 +444,13 @@ class _CursesWindow:
...
+    def getstr(self) -> None: ...
     @overload
-    def getstr(self, n: int) -> _chtype: ...
+    def getstr(self, n: int) -> None: ...
     @overload
-    def getstr(self, y: int, x: int) -> _chtype: ...
+    def getstr(self, y: int, x: int) -> None: ...
     @overload
-    def getstr(self, y: int, x: int, n: int) -> _chtype: ...
+    def getstr(self, y: int, x: int, n: int) -> None: ...
```
Comment (Akuli) [inline stdlib/_curses.pyi:453]: `getstr()` returns bytes: https://github.com/python/cpython/blob/46c8d915715aa2bd4d697482aa051fe974d440e1/Modules/_cursesmodule.c#L1600

File: stdlib/_curses.pyi [hunk 2]
```diff
@@ -460,9 +460,9 @@ class _CursesWindow:
     def idlok(self, yes: bool) -> None: ...
     def immedok(self, flag: bool) -> None: ...
     @overload
-    def inch(self) -> _chtype: ...
+    def inch(self) -> bytes: ...
     @overload
-    def inch(self, y: int, x: int) -> _chtype: ...
+    def inch(self, y: int, x: int) -> bytes: ...
```
Comment (Akuli) [inline stdlib/_curses.pyi:465]: `inch()` returns int (in C it returns `unsigned long`, which gets converted to Python `int`): https://github.com/python/cpython/blob/46c8d915715aa2bd4d697482aa051fe974d440e1/Modules/_cursesmodule.c#L1708

File: stdlib/_curses.pyi [hunk 3]
```diff
@@ -478,9 +478,9 @@ class _CursesWindow:
     @overload
     def insstr(self, y: int, x: int, str: str, attr: int = ...) -> None: ...
     @overload
-    def instr(self, n: int = ...) -> _chtype: ...
+    def instr(self, n: int = ...) -> str: ...
```
Comment (Akuli) [inline stdlib/_curses.pyi:481]: This overload should return `bytes` too. Both overloads of `instr()` use this return statement: https://github.com/python/cpython/blob/46c8d915715aa2bd4d697482aa051fe974d440e1/Modules/_cursesmodule.c#L1785

```suggestion
    def instr(self, n: int = ...) -> bytes: ...
```

**Comment by Rohan-Salwan** [comment]
@Akuli thanks for guiding me.

**Review by Akuli — APPROVED** [review]

**Comment by Akuli** [comment]
Thanks!

**Pull request merged**